src/main/java/com/learning/TestsConfiguration.java

```diff
@@ -0,0 +1,11 @@
+package com.learning;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+
+@Configuration
+@ComponentScan
```

Comment:
Please narrow down component scan scope using `basePackage` property.

Reply:
With blank `basePackage` property by default current Class package and all subpackages will be taken into consideration for beans scanning. That is why I put this class in `com.learning` package. 
I don't see any possibility to narrow it down since then I have `components` in few directories.